document/core/syntax/conventions.rst

```diff
@@ -123,11 +123,9 @@ Vectors
 
 *Vectors* are bounded sequences of the form :math:`A^n` (or :math:`A^\ast`),
 where the :math:`A` can either be values or complex constructions.
-A vector can have at most :math:`2^{32}-1` elements.
 
 .. math::
    \begin{array}{lllll}
    \production{vector} & \vec(A) &::=&
-     A^n
-     & (\iff n < 2^{32})\\
```

Comment:
Ah, removing this constraint is not correct, since vectors are used in many places throughout the spec that continue to be syntactically restricted to 32-bit length.

The correct fix is to change the content types of table and memory instances from vectors to unrestricted sequences `xyz*`.

Reply:
Done. As far as I can tell, nothing else in the spec needs to change to support this, since constructs like `|foo|` and `foo[a:b]` still work with sequences.